sql/core/src/main/scala/org/apache/spark/sql/execution/arrow/ArrowWriter.scala

```diff
@@ -343,3 +348,36 @@ private[arrow] class StructWriter(
     children.foreach(_.reset())
   }
 }
+
+private[arrow] class MapWriter(val valueVector: MapVector,
+                               val keyWriter: ArrowFieldWriter,
+                               val valueWriter: ArrowFieldWriter) extends ArrowFieldWriter {
+
+  override def setNull(): Unit = {}
```

Comment:
can't we set null for a map vector?

Reply:
map vector implemented as an extension of list vector. it is represented as list of struct. it seems we can not set key as null. I found following comments in code,  
```
if a map is set at a given index, there must be an entry. In other words, the StructVector data is non-nullable. 
```

- https://github.com/apache/arrow/blob/master/java/vector/src/main/java/org/apache/arrow/vector/complex/MapVector.java#L36